django/http/request.py

```diff
@@ -374,13 +393,10 @@ def __init__(self, query_string=None, mutable=False, encoding=None):
                 except UnicodeDecodeError:
                     # ... but some user agents are misbehaving :-(
                     query_string = query_string.decode('iso-8859-1')
-            for key, value in parse_qsl(query_string or '',
-                                        keep_blank_values=True,
-                                        encoding=encoding):
+            for key, value in limited_parse_qsl(query_string, **parse_qsl_kwargs):
                 self.appendlist(key, value)
         else:
-            for key, value in parse_qsl(query_string or '',
-                                        keep_blank_values=True):
+            for key, value in limited_parse_qsl(query_string, **parse_qsl_kwargs):
```

Comment:
The behavior re. `encoding` appears to have changed here, possibly unintentionally.
(Ie now we'd be passing an 'encoding' argument at this point. where we wouldn't before)